core/src/main/java/com/exadel/etoolbox/linkinspector/core/services/data/impl/UserConfigImpl.java

```diff
@@ -0,0 +1,95 @@
+package com.exadel.etoolbox.linkinspector.core.services.data.impl;
+
+import com.exadel.etoolbox.linkinspector.core.services.data.UserConfig;
+import com.exadel.etoolbox.linkinspector.core.services.helpers.RepositoryHelper;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ValueMap;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+@Component(service = UserConfig.class)
+@Slf4j
+public class UserConfigImpl implements UserConfig {
+
+    @Reference
+    private RepositoryHelper repositoryHelper;
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T extends Annotation> T apply(T baseConfig) {
+        Handler<T> handler = new Handler<>(baseConfig, getProperties(baseConfig.annotationType().getName()));
+        Object proxy = Proxy.newProxyInstance(
+                baseConfig.annotationType().getClassLoader(),
+                new Class[]{baseConfig.annotationType()},
+                handler);
+        return (T) proxy;
+    }
+
+    private Map<String, Object> getProperties(String configId) {
+        try(ResourceResolver resourceResolver = repositoryHelper.getServiceResourceResolver()){
+            Resource resource = resourceResolver.getResource("/conf/etoolbox-link-inspector/data/optionsconfig/" + StringUtils.substringBeforeLast(configId, "$"));
```

Comment:
❗ This is the `CONFIG_PATH` constant from the previous class. Even it it weren't, it would _have to_ be declared as a public constant in some very well known place. Becasuse such things as paths are subject to change. In fact, this very path will need to be changed _very soon_. Because what have we discussed twice already?)) That this is going to be `/conf/etoolbox/link-inspector` ))

Reply:
done